crates/core/src/commands/restore.rs

```diff
@@ -696,9 +698,12 @@ impl RestorePlan {
             };
             let length = bl.data_length();
 
+            let usize_lenth =
+                usize::try_from(length).map_err(CommandErrorKind::ConversionFromIntFailed)?;
```

Comment:
This will never fail unless we are on a 16 bit machine. `length` is basically a `u32` converted into a `u64`. 
But maybe just keep it, however maybe rename into `usize_length`?